### ISSUE TITLE
Abstract or static attributes with the official UML syntax using `name: Type`

### ISSUE THREAD

**Issue by AlexanderGrasberger** [mermaid-js/mermaid#3001]
**Is your feature request related to a problem? Please describe.**
I really like the `classDiagram` Feature. We use it in university for our students to quickly and easily make some UML class diagrams for their projects. The only problem is that we can't use the official UML syntax for attributes like this: `-name: String` in conjunction with the **abstract** `*` and **static** `$` flags. This works for an attribute that is build like this: `-String name` but not for `-name: String`.

**Describe the solution you'd like**
I think a useful tool like this should support the official UML attribute syntax.
A possible syntax could look like this `-name1: String*` or maybe this `-name1*: String`

**Additional context**
Here is an example of the current behavior of mermaid:

```
classDiagram
    class Example {
        -name1: String*
        -name2: String$
        -name1*: String
        -name2$: String
        -String name1*
        -String name2$
    }
```
```mermaid
classDiagram
    class Example {
        -name: String*
        -name: String$
        -name*: String
        -name$: String
        -String name*
        -String name$
    }
```

**Comment by fer-rum** [comment]
I was just about to raise the same issue. There is no problem with writing it down as `String name$` in _mermaid_ but it should be rendered appropriately.

Furthermore: If one does an analysis class diagram, usually no type is given, but `name$` will just be rendered verbatim, which is no problem for regular attributes, but an issue with static attributes (The static modifier will be included as is, instead the attribute should be underlined).

**Comment by skogsbaer** [comment]
I open the same issue: #1953
It with fixed by @Xstoudi in PR #2269 but apperantly it only works in certain cases. @Dampfwalze mentioned this already in the original PR #2269

**Comment by skogsbaer** [comment]
@AlexanderGrasberger  Please re-open, there is now a PR #3855

**Comment by jgreywolf** [comment]
@AlexanderGrasberger I took the code from your PR and merged with PR #4303 I just submitted - we hit alot of the same areas - especially in the tests, and this should speed things up a bit
